dev-docs/e2e/bare-metal-runner.md

```diff
@@ -1,5 +1,221 @@
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+
+# Download docker package
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+```
+
+Install k9s (optional)
+```bash
+curl -fsL https://github.com/derailed/k9s/releases/latest/download/k9s_Linux_amd64.tar.gz | tar -C /usr/local/bin -xz k9s
+```
+
+Install helm
+```bash
+curl -fsL https://get.helm.sh/helm-v3.17.1-linux-amd64.tar.gz | tar -C /tmp -xz linux-amd64/helm && mv /tmp/linux-amd64/helm /usr/local/bin
+```
+
+Install K3s
+```bash
+curl -sfL https://get.k3s.io | sh -
+```
+
+Install Longhorn into K3s
+```bash
+helm repo add longhorn https://charts.longhorn.io
+helm repo update
+helm install longhorn longhorn/longhorn --namespace longhorn-system --create-namespace
+```
+
+## Networking
+Add the device to the tailscale network.
+For this you have to have admin privileges, if you don't see the
+overview of machines when visiting
+https://login.tailscale.com/admin/machines, notify another engineer.
+
+On this page, click "Add device." In the settings add the "ssh-access"
+label. This is needed since all engineers have the "devs"
+role, which allows them to ssh into all devices that have the "ssh-access"
+tag.
+Follow the other instructions on the tailscale website to add the device
+and execute the given script on the machine.
+After the installation, execute:
+```bash
+sudo tailscale up --ssh
+```
+
+
+Add a firewall for incoming connections if the server is reachable via
+a public IP:
+```bash
+ufw status
+ufw app list
+ufw allow 22
+ufw allow OpenSSH
+ufw show added
+ufw enable
+```
+
+## Add server as a GitHub runner
+
+First, create another user, which the runner service will use.
+```bash
+useradd -s /bin/bash -m -G sudo,docker github
+```
+
+Put the K3s kubeconfig into the default dir for the user:
+```bash
+mkdir -p /home/github/.kube
+chmod +r /etc/rancher/k3s/k3s.yaml
+ln -s /etc/rancher/k3s/k3s.yaml /home/github/.kube/config
+sudo chown -c github /home/github/.kube/config
+```
+
+The CI jobs build things with nix, therefore install it with the
+Determinate Nix Installer:
+https://github.com/DeterminateSystems/nix-installer?tab=readme-ov-file#determinate-nix-installer
+
+
+Check what filesystem the server has:
+```bash
+findmnt /
+```
+
+If it's anything other than a btrfs, setup a btrfs builder volume.
+The instructions are taken from https://github.com/edgelesssys/contrast/blob/a62af98f2df761116109310a6af4adcb66e758c0/.github/actions/setup_nix/action.yml#L35.
+Note that those instructions have to be executed every time the server is restarted.
```

Comment:
This should not be the case. Why do you think it is?

Reply:
Yeah it should be a systemd unit, I can have a look into this.

Reply:
I'd advocate for keeping things simple by just adding a line to `/etc/fstab` and relying on `systemd-fstab-generator`, which parses the entries and converts them to systemd mount units :)

Reply:
done